clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10761,7 +10761,9 @@ def err_builtin_launder_invalid_arg : Error<
 // SYCL-specific diagnostics
 def err_sycl_kernel_incorrectly_named : Error<
   "kernel %select{name is missing"
-  "|needs to have a globally-visible name}0">;
+  "|needs to have a globally-visible name"
+  "|name cannot be templated using unscoped enum without fixed underlying type"
```

Comment:
text is a little clunky.... anyone have a better suggestion?

Reply:
I will greatly appreciate that :) I don't know what I hate more - figuring out variable names or diagnostic messages

Reply:
I changed this to - kernel name is invalid. Unscoped enum requires fixed underlying type.